app/helpers/application_html_formatters_helper.rb

```diff
@@ -13,15 +13,32 @@ module ApplicationHTMLFormattersHelper
 
   # List of video hosting site URLs to allow
   VIDEO_URL_WHITELIST = Regexp.union(
-    /\A(?:https?:)?\/\/(?:www\.)?youtube\.com\//,
+    /\A(?:https?:)?\/\/(?:www\.)?(?:m.)?youtube\.com\//,
     /\A(?:https?:)?\/\/(?:www\.)?youtu.be\//,
-    /\A(?:https?:)?\/\/(?:www\.)?vimeo\.com\//,
+    /\A(?:https?:)?\/\/(?:www\.)?(?:player.)?vimeo\.com\//,
     /\A(?:https?:)?\/\/(?:www\.)?vine\.co\//,
     /\A(?:https?:)?\/\/(?:www\.)?instagram\.com\//,
-    /\A(?:https?:)?\/\/(?:www\.)?dailymotion\.com\//,
+    /\A(?:https?:)?\/\/(?:www\.)?(?:geo.)?dailymotion\.com\//,
+    /\A(?:https?:)?\/\/(?:www\.)?dai\.ly\//,
     /\A(?:https?:)?\/\/(?:www\.)?youku\.com\//
   )
 
+  OEMBED_WHITELIST_TRANSFORMER = lambda do |env|
+    node, node_name = env[:node], env[:node_name]
+
+    return if env[:is_whitelisted] || !node.element?
+
+    return unless node_name == 'oembed'
+    return unless node['url']&.match VIDEO_URL_WHITELIST
+
+    resource = OEmbed::Providers.get(node['url'])
+    new_node = Nokogiri::HTML5.fragment(resource.html).children.first
+
+    node.add_next_sibling(new_node)
```

Comment:
This works when saving the text. When the data is sent to the frontend, this is run again, would this create a duplicate `<iframe/>`?

Reply:
That's right. CKEditor will strip the `<iframe>`s when `onChange` is triggered. Most of our forms disable their submit buttons if the form is unchanged. And for anything to change in CKEditor, `onChange` must be called.

Therefore, these premises conclude that there must be no case where there will be duplicate `<iframe>`s, to the best of my knowledge.

Reply:
Yup there shouldnt be duplicate from FE to BE then. What about BE to FE?

Reply:
How would the back-end have `<iframe>`s saved in the database in the first place?

Ideally, we should save no `<iframe>`s to the database, and only `<oembed>`s.